guacamole/src/main/frontend/src/app/clipboard/directives/guacClipboard.js

```diff
@@ -92,12 +119,20 @@ angular.module('clipboard').directive('guacClipboard', ['$injector',
         // that indicate the clipboard field may have been changed
         element.addEventListener('input', updateClipboardData);
         element.addEventListener('change', updateClipboardData);
+        element.addEventListener('focus', displayAndUpdateClipboardEditor);
+
 
         // Update remote clipboard if local clipboard changes
         $scope.$on('guacClipboard', function clipboardChanged(event, data) {
             updateClipboardEditor(data);
         });
 
+        // Update the prefocus clipboard editor message
+        guacTranslate('CLIENT.TEXT_CLIPBOARD_AWAITING_FOCUS', '').then(
```

Comment:
It might be a bit cleaner/easier to just do the translation in the template. That would also improve separation of concerns - the controller shouldn't have to care about what specific messages are being displayed to the user. All it should need to know is what the state of the clipboard is, exposing that state to the template which can then render different elements (with translated text in them) according to that state.

Reply:
Also, what if a user literally has the text "Focus here to view clipboard data..." in their clipboard?

 Having some quick visual method to clearly identify that the clipboard is inactive would be nice. Rather than just sticking a message into the textarea - how about visually disabling (greying out, etc) the textarea and rendering a message in a different spot where it couldn't be confused with actual clipboard contents?